--- src/Purust/CodeGen.purs
+++ src/Purust/CodeGen.purs
@@ -1071,7 +1071,8 @@
                                    Nothing -> Any
                     valTy = inferTypeExprGlobal currentMod aritiesMap globalClassFields bound val
                     resCode = boxUnbox currentMod expectedTy valTy valCode
-                in sanitizeIdent fieldName <> ": " <> resCode
+                    debugComment = "/* expectedTy: " <> printType expectedTy <> " valTy: " <> printType valTy <> " expStr: " <> codegenExprType currentMod true expectedTy <> " actStr: " <> codegenExprType currentMod true valTy <> " */ "
+                in sanitizeIdent fieldName <> ": " <> debugComment <> resCode
           ) fields)
         in "std::rc::Rc::new(" <> structName <> " { " <> structFieldsCode <> " })"
       Nothing ->
