--- src/Purust/CodeGen.purs
+++ src/Purust/CodeGen.purs
@@ -1071,7 +1071,8 @@
                                    Nothing -> Any
                     valTy = inferTypeExprGlobal currentMod aritiesMap globalClassFields bound val
                     resCode = boxUnbox currentMod expectedTy valTy valCode
-                in sanitizeIdent fieldName <> ": " <> resCode
+                    _ = if structName == "Purs_Data_Semigroup::Semigroup" then Debug.trace ("SEMIGROUP field=" <> fieldName <> " expectedTy=" <> codegenExprType currentMod true expectedTy <> " valTy=" <> codegenExprType currentMod true valTy <> " valCode=" <> valCode <> " resCode=" <> resCode) \_ -> unit else unit
+                in sanitizeIdent fieldName <> ": " <> resCode
           ) fields)
         in "std::rc::Rc::new(" <> structName <> " { " <> structFieldsCode <> " })"
       Nothing ->
